ArcGISRuntimeSDKQt_CppSamples/EditData/SnapGeometryEdits/SnapGeometryEdits.cpp

```diff
@@ -0,0 +1,335 @@
+{
+  m_geometryEditor->snapSettings()->setEnabled(snappingCheckedState);
+}
+
+// Toggles snapping for the point layer snap source at [index] using the checked value from the snap settings
+void SnapGeometryEdits::pointSourceEnabledStatus(bool snappingCheckedState, int index)
+{
+  m_pointSourceCheckedState[index] = snappingCheckedState;
+  m_pointSourceList[index]->setEnabled(m_pointSourceCheckedState[index]);
+
+}
+
+// Toggles snapping for the polyline layer snap source at [index] using the checked value from the snap settings
+void SnapGeometryEdits::polylineSourceEnabledStatus(bool snappingCheckedState, int index)
+{
+  m_polylineSourceCheckedState[index] = snappingCheckedState;
+  m_polylineSourceList[index]->setEnabled(m_polylineSourceCheckedState[index]);
+
+}
+
+// Enable snapping for all the point layer snap sources
+void SnapGeometryEdits::onPointLayersEnabled()
+{
+  for (int index = 0; index < m_pointSourceList.size(); ++index)
+  {
+    pointSourceEnabledStatus(true, index);
+  }
+  emit pointSourceCheckedStateChanged();
+}
+
+// Enable snapping for all the polyline layer snap sources
+void SnapGeometryEdits::onPolylineLayersEnabled()
+{
+  for (int index = 0; index < m_polylineSourceList.size(); ++index)
+  {
+    polylineSourceEnabledStatus(true, index);
+  }
+  emit polylineSourceCheckedStateChanged();
+}
+
+// Starts the GeometryEditor using the selected geometry type
+void SnapGeometryEdits::startEditor(GeometryEditorMode geometryEditorMode)
+{
+  switch (geometryEditorMode)
+  {
+  case GeometryEditorMode::PointMode:
+    m_geometryEditor->start(Esri::ArcGISRuntime::GeometryType::Point);
+    break;
+
+  case GeometryEditorMode::MultipointMode:
+    m_geometryEditor->start(Esri::ArcGISRuntime::GeometryType::Multipoint);
+    break;
+
+  case GeometryEditorMode::PolylineMode:
+    m_geometryEditor->start(Esri::ArcGISRuntime::GeometryType::Polyline);
+    break;
+
+  case GeometryEditorMode::PolygonMode:
+    m_geometryEditor->start(Esri::ArcGISRuntime::GeometryType::Polygon);
+    break;
+
+  default:
+    break;
+  }
+  emit geometryEditorStartedChanged();
+}
+
+// Stops the GeometryEditor and append the new graphic to the graphics overlay
+void SnapGeometryEdits::stopEditor()
+{
+  const Geometry geometry = m_geometryEditor->stop();
+  emit geometryEditorStartedChanged();
+
+  if (m_editingGraphic)
+  {
+    m_editingGraphic->setGeometry(geometry);
+
+    m_editingGraphic->setVisible(true);
+    m_editingGraphic = nullptr;
+
+    return;
+  }
+
+  Symbol* geometrySymbol = nullptr;
+  switch (geometry.geometryType())
+  {
+  case GeometryType::Point:
+    geometrySymbol = m_pointSymbol;
+    break;
+
+  case GeometryType::Multipoint:
+    geometrySymbol = m_multiPointSymbol;
+    break;
+
+  case GeometryType::Polyline:
+    geometrySymbol = m_lineSymbol;
+    break;
+
+  case GeometryType::Polygon:
+    geometrySymbol = m_polygonSymbol;
+    break;
+
+  default:
+    return;
+  }
```

Comment:
I would consider refactoring this block into a private helper function.

Reply:
done